### PR TITLE
PUBDEV-5537: Enforce system-level command line arguments in h2odriver (eg. security)

### PR DESCRIPTION
**Pull request by michalkurka**

### CODE REVIEW

File: h2o-hadoop/h2o-mapreduce-generic/src/main/java/water/hadoop/h2odriver.java [hunk 1]
```diff
@@ -40,6 +39,9 @@
  */
 @SuppressWarnings("deprecation")
 public class h2odriver extends Configured implements Tool {
+
+  final static String SYS_ARGS_FILE = "/etc/h2odriver.args";
```
Comment (tomkraljevic) [inline h2o-hadoop/h2o-mapreduce-generic/src/main/java/water/hadoop/h2odriver.java]: I would prefer
/etc/h2o/h2odriver.args

**Comment by tomkraljevic** [comment]
The argument injection in steam is done in four phases:

hadoop early args ("defaults")
hadoop late args ("overrides")

h2odriver early args ("defaults")
h2odriver late args ("overrides")

this matrix solves lots of useful combinations.

**Comment by michalkurka** [comment]
@tomkraljevic it will impact SW external cluster if the configuration file is present on the edge node that is submitting the external cluster job. There can be a clash if both SW and the system configuration specifies the same argument.

It is not well suited for this kind of deployment, it shouldn't be used in that case. Or the user has to be well aware of what SW is doing.

**Comment by jakubhava** [comment]
@michalkurka I was thinking about it and  I would probably prefer to have option to disable this behavior, but that option would need to be in H2O itself.

**Comment by michalkurka** [comment]
@jakubhava I added an option to bypass the default arguments by specifying alternative "argument configuration":

    hadoop jar h2odriver.jar -Dai.h2o.args.config=sw-external -mapperXmx 3g -nodes 1

This option becomes part of the job configuration and it is easy to see in history that user bypassed the default config.

If the alternative configuration is specified and the corresponding argument file doesn't exist it checks whether the default file exists. If the default file exists it shows an error:

    ERROR: There is no arguments file for configuration 'sw-external', however, the arguments file exists for the default configuration.        
            Please create an arguments file also for configuration 'sw-external' and store it in '/etc/h2o/sw-external.args' (the file can be empty).

This makes sure that the defaults cannot be arbitrarily bypassed, the corresponding file has to exist if the default file exists. It also limits the chance of making a mistake in specifying the configuration.

**Comment by jakubhava** [comment]
Perfect, thanks @michalkurka ! I will update Sparkling Water based on this change when it's inside

**Pull request merged**